src/ServiceInsight/MessageList/MessageListViewModel.cs

```diff
@@ -97,28 +97,49 @@ public void CopyHeaders()
             clipboard.CopyTo(generalHeaderDisplay.HeaderContent);
         }
 
-        public void RefreshMessages(string orderBy = null, bool ascending = false)
+        public void RefreshMessages(string link, int pageIndex)
         {
-            var serviceControlExplorerItem = selectedExplorerItem as ServiceControlExplorerItem;
-            if (serviceControlExplorerItem != null)
+            using (workNotifier.NotifyOfWork("Loading messages..."))
             {
-                RefreshMessages(searchQuery: SearchBar.SearchQuery,
-                                     endpoint: null,
-                                     orderBy: orderBy,
-                                     ascending: ascending);
+                var pagedResult = serviceControl.GetAuditMessages(link);
+                if (pagedResult == null)
+                {
+                    return;
+                }
+
+                pagedResult.CurrentPage = pageIndex;
+                TryRebindMessageList(pagedResult);
+
+                SearchBar.IsVisible = true;
+                SearchBar.SetupPaging(pagedResult.TotalCount > 0 ? pagedResult.CurrentPage : 0, pagedResult.TotalCount, pagedResult.PageSize, pagedResult.Result,
+                    pagedResult.NextLink,
+                    pagedResult.PrevLink,
+                    pagedResult.FirstLink,
+                    pagedResult.LastLink,
+                    link);
             }
+        }
 
+        public void RefreshMessages(string orderBy = null, bool ascending = false)
+        {
             var endpointNode = selectedExplorerItem as AuditEndpointExplorerItem;
             if (endpointNode != null)
             {
-                RefreshMessages(searchQuery: SearchBar.SearchQuery,
+                InitSearch(searchQuery: SearchBar.SearchQuery,
                                      endpoint: endpointNode.Endpoint,
                                      orderBy: orderBy,
                                      ascending: ascending);
             }
+            else
+            {
+                InitSearch(searchQuery: SearchBar.SearchQuery,
+                    endpoint: null,
+                    orderBy: orderBy,
+                    ascending: ascending);
+            }
```

Comment:
the branching seem to be very similar and duplicated? could we get away with one? e.g.

endpoint = endpointNode ? null : endpointNode.Endpoint;
InitSearch(....)

so no branching.